src/osd/PrimaryLogPG.cc

```diff
@@ -7365,6 +7373,8 @@ inline int PrimaryLogPG::_delete_oid(
     interval_set<uint64_t> ch;
     ch.insert(0, oi.size);
     ctx->modified_ranges.union_of(ch);
+    ctx->clean_regions.mark_data_region_dirty(0, oi.size);
+    ctx->clean_regions.mark_omap_dirty();
```

Comment:
could use a comment here about why xattrs aren't mentioned.
if I understand correctly, this handles xattrs correctly because during recovery we either 1) delete the object (if there are no subsequent writes) or 2) when recovering subsequent writes to the object, always remove and then copy all the xattrs

Reply:
yeah,  it is.